src/oop/special_class_method/hard/q5/MainTest.java

```diff
@@ -0,0 +1,260 @@
+        _assertEquals(value2, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoIntInputProvider")
+    public void roadClassHasCorrectSetMaxCapacityMethod(int initialValue, int setValue) throws Throwable {
+        Object[][] arguments = {
+                {2, int.class},
+                {"Initial String", String.class},
+                {initialValue, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, int.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setInt2MethodName, "does not correctly set the value of the", int2FieldName, "field.");
+        Object setMethodOutput = testClass.callMethod(setInt2MethodName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(setValue, testClass.getFieldValue(classInstance, int2FieldName), incorrectSetMethodMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setInt2MethodName, "should not return any output"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void roadClassHasCorrectGetMaxCapacityMethod(int value1, String value2, int value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class},
+                {value3, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] getMethodModifiers = {"public"};
+        String incorrectGetMethodMessage = String.join(" ",
+                "Your", getInt2MethodName, "does not correctly get the value of the", int2FieldName, "field.");
+        Object getMethodOutput = testClass.callMethod(getInt2MethodName, getMethodModifiers, classInstance);
+        _assertEquals(value3, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    @ParameterizedTest
+    @MethodSource("constructorInputProvider")
+    public void chairClassHasCorrectToStringMethod(int value1, String value2, int value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class},
+                {value3, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        String[] methodModifiers = {"public"};
+        String expected = className + "{" + intFieldName + ": " + value1 + ", " + StringFieldName + ": " + value2 + ", " + int2FieldName + ": " + value3 + "}";
+        String incorrectToStringMessage = String.join(" ",
+                "Your", className, " toString method does not return the correct String.");
+        Object output = testClass.callMethod("toString", methodModifiers, classInstance);
+        _assertEquals(expected, output, incorrectToStringMessage);
+    }
+
```

Comment:
There should be another test that checks if all the methods work together properly